packages/fast-components-react-base/src/select/select.tsx

```diff
@@ -364,13 +361,19 @@ class Select extends Foundation<SelectHandledProps, SelectUnhandledProps, Select
         if (this.props.disabled || e.defaultPrevented) {
             return;
         }
+
         switch (e.keyCode) {
             case KeyCodes.enter:
             case KeyCodes.space:
+                e.preventDefault();
                 this.toggleMenu(!this.state.isMenuOpen);
+                if (this.state.isMenuOpen && this.props.isMenuOpen !== true) {
```

Comment:
Should this `if (this.state.isMenuOpen && this.props.isMenuOpen !== true)` be using the `validateMenuState ` method instead?

Reply:
Yep, that worked.